app/scripts/modules/core/src/managed/ArtifactDetail.tsx

```diff
@@ -26,28 +21,95 @@ function shouldDisplayResource(name: string, type: string, resource: IManagedRes
   return !!resource.moniker && name === resource.artifact?.name && type === resource.artifact?.type;
 }
 
-const ConstraintCards = memo(
+const inStyles = {
+  opacity: 1,
+  transform: 'scale(1.0, 1.0)',
+};
+
+const outStyles = {
+  opacity: 0,
+  transform: 'scale(0.95, 0.95)',
+};
+
+const cardTransitionConfig = {
+  from: outStyles,
+  // KLUDGE: all we're *actually* doing in this scary looking handler
+  // is delaying the start of any enter transitions for a fixed time
+  // so parent transitions for the overall layout have time to start.
+  // Unfortunately today useTransition doesn't support fixed delays
+  // without tapping into this promise-based orchestration feature (ew).
+  // When react-spring v9 is released, this can be changed
+  // to a function that returns { to: inStyles, delay: 180 }
+  enter: () => async (next: (_: React.CSSProperties) => any) => {
+    await new Promise(resolve => setTimeout(resolve, 180));
+    next(inStyles);
+  },
+  leave: outStyles,
+  trail: 40,
+  config: { mass: 1, tension: 600, friction: 40 },
+} as UseTransitionProps<JSX.Element, React.CSSProperties>;
+
+type IEnvironmentCardsProps = Pick<IArtifactDetailProps, 'application' | 'version' | 'allVersions'> & {
+  environment: IManagedArtifactSummary['versions'][0]['environments'][0];
+};
+
+const EnvironmentCards = memo(
```

Comment:
This whole block of logic w/ `EnvironmentCards` and related stuff is kind of gross and messy, though certainly more organized that it was when I first wrote it. I'm leaving it as-is for now because I want to confirm that this transition is something useful we want to keep. If we decided it is, I'm going to extract a `<CardList />` component that accepts an array of JSX elements and whose _only_ job is to add list margin + do enter/leave transitions. IMO that will clean this code up immensely and likely eliminate the need for this component (or at least heavily simplify it)